File: Toy_Language_Interpreter/Model/Statements/PrintStatement.java
package Model.Statements;

import ADTs.IDictionary;
import ADTs.IList;
import Exceptions.DivisionByZeroException;
import Exceptions.HeapReadingException;
import Exceptions.UndefinedOperationException;
import Exceptions.UndefinedVariableException;
import Heap.IHeap;
import Model.Expressions.*;
import Model.ProgramState;

public class PrintStatement implements IStatement {
    private IExpression exp;

    public PrintStatement(IExpression ex) {
        this.exp = ex;
    }

    public String toString() {
        return "Print(" + exp.toString() + ")";
    }

    public ProgramState execute(ProgramState state) throws UndefinedVariableException, DivisionByZeroException, UndefinedOperationException, HeapReadingException {
        IList<Integer> out = state.getOut();
        IDictionary<String, Integer> symTable = state.getSymTable();
        IHeap heap = state.getHeap();
        Integer result = this.exp.eval(symTable, heap);
        out.addFirst(result);
        return state;
    }

}

File: Toy_Language_Interpreter_C/Statements/PrintStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class PrintStatement : IStatement
    {
        private IExpression expression;

        public PrintStatement(IExpression exp)
        {
            this.expression = exp;
        }

        public ProgramState Execute(ProgramState state)
        {
            try
            {
                int result = this.expression.Eval(state.SymTable);
                state.OutTable.AddFirst(result);
                return state;
            }
            catch(DivideByZeroException)
            {
                throw new DivideByZeroException("DIVISION BY ZERO. Cannot divide by 0.");
            }
            catch(UndefinedVariableException)
            {
                throw new UndefinedVariableException();
            }
            catch(UndefinedOperationException)
            {
                throw new UndefinedOperationException();
            }
        }

        public override string ToString()
        {
            return "Print(" + this.expression.ToString() + ")";
        }
    }
}
